src/plugins/formulas/formulas.js

```diff
@@ -203,36 +250,231 @@ export class Formulas extends BasePlugin {
       });
     }
 
+    const hfSetUpProperly = this.setupHF();
+
+    if (!hfSetUpProperly) {
+      return;
+    }
+
+    // HyperFormula events:
+    this.hyperformula.on('valuesUpdated', (...args) => this.onHFvaluesUpdated(...args));
+
+    this.applyHFSettings();
+
     super.enablePlugin();
   }
 
   /**
    * Disables the plugin functionality for this Handsontable instance.
    */
   disablePlugin() {
-    // TODO add tests for this line
-    this.hyperformula.destroy();
-
     super.disablePlugin();
   }
 
+  /**
+   * Triggered on `updateSettings`.
+   */
+  updatePlugin() {
+    this.#settings = this.hot.getSettings()[PLUGIN_KEY];
+
+    this.applyHFSettings();
+
+    if (isDefined(this.#settings.sheetName) && this.#settings.sheetName !== this.sheetName) {
+      this.switchSheet(this.#settings.sheetName);
+    }
+
+    super.updatePlugin();
+  }
+
   /**
    * Destroys the plugin instance.
    */
   destroy() {
+    if (this.hyperformula) {
+      const hfInstances = staticRegister('formulas').getItem('hyperformulaInstances');
+      const sharedHFInstanceUsage = hfInstances ? hfInstances.get(this.hyperformula) : null;
+
+      if (sharedHFInstanceUsage && sharedHFInstanceUsage.includes(this.hot.guid)) {
+        sharedHFInstanceUsage.splice(
+          sharedHFInstanceUsage.indexOf(this.hot.guid),
+          1
+        );
+
+        if (sharedHFInstanceUsage.length === 0) {
+          hfInstances.delete(this.hyperformula);
+          this.hyperformula.destroy();
+        }
+      }
+    }
+
     super.destroy();
   }
 
+  /**
+   * Add a sheet to the shared HyperFormula instance.
+   *
+   * @param {Array} sheetData Data passed to the shared HyperFormula instance. Has to be declared as an array of
+   * arrays - array of objects is not supported in this scenario.
+   * @param {string} [sheetName] The new sheet name. If not provided, will be auto-generated by HyperFormula.
+   * @param {boolean} [autoLoad] If `true`, the new sheet will be loaded into the Handsontable instance.
+   * @returns {boolean} `false` if the data format is unusable, `true` otherwise.
+   */
+  addSheet(sheetData, sheetName, autoLoad) {
```

Comment:
It's probably just my lack of experience showing, but why have `autoLoad` if it's just a single line of code on the user side?

Reply:
I think it's just personal preference. I figured, that if `addSheet` is a method on _Handsontable_, it should not only add the sheet to HF, but also to the _Handsontable instance_, or at least give the user the option to do that.

Reply:
I think this is breaking our convention? Didn't seen such "modifier" argument anywhere else in the code base.
And when HF introduce an extension to addSheets https://github.com/handsontable/hyperformula/issues/643 we will have arguments mismatch.

Or we can implement namedExpressions right now inside this method
```ts
addSheet(sheetName, sheetData, namedExpressions, autoLoad) {
  namedExpressions.forEach(nE => this.engine.addNamedExpression(nE))
 this.engine.setSheetData(sheetName, sheetData)
}
```

Reply:
It wasn't meant to mirror the HF method's arguments though? I mean, if it should - it will.

As for the convention, I think it's similar to arguments for rendering/not rendering after actions, such as:
- [plugins/comments/comments.js#L306](https://github.com/handsontable/handsontable/blob/master/src/plugins/comments/comments.js#L306)
- [plugins/nestedRows/data/dataManager.js#L538](https://github.com/handsontable/handsontable/blob/master/src/plugins/nestedRows/data/dataManager.js#L538)

etc

Reply:
It meant, but HF will change the method, we may do that now or break later.

Oh I see...

Reply:
But then again, in other places we require manual `render`s, so I'm ok with removing it ;)

Reply:
🤯 Done then!